core/kazoo_config/src/kapps_config_util.erl

```diff
@@ -80,3 +81,14 @@ account_schema(Config) when is_binary(Config) ->
     Name = account_schema_name(Config),
     {'ok', Schema} = kz_json_schema:load(Name),
     Schema.
+
+-spec system_config_document_schema(ne_binary()) -> kz_json:object().
+system_config_document_schema(Id) ->
+    Flat = [
```

Comment:
> objects should have keys as lists of keys

Well, it works both ways (e.g. one can have a key as binary() or [binary()], I'm not quite sure we need to restrict this. And this behaviour comes from how set_value/3 is defined.